### PR TITLE
Dispute path

### PR DESCRIPTION
**Pull request by bpierre**
Builds on https://github.com/aragon/court-dashboard/pull/30

### CODE REVIEW

File: src/components/Disputes/DisputeDetail.js [hunk 1]
```diff
@@ -1,35 +1,53 @@
...
+import { useHistory } from 'react-router-dom'
+import useDisputesSubscription from './hooks/useDisputesSubscription'
 import DisputeInfo from './DisputeInfo'
 import DisputeEvidences from './DisputeEvidences'
 import DisputeTimeline from './DisputeTimeline'
 
-function DisputeDetail({ dispute, onBack }) {
+function DisputeDetail({ match }) {
+  const history = useHistory()
+  const disputes = useDisputesSubscription()
```
Comment (fabriziovigevani) [inline src/components/Disputes/DisputeDetail.js]: Should we do a dispute subscription by the id  instead of subscribing to all disputes?

Comment (bpierre) [reply src/components/Disputes/DisputeDetail.js]: Done: https://github.com/aragon/court-dashboard/pull/41/commits/c6e04591323e75c22cdbc55aff4acfc5d2aa79d5 and https://github.com/aragon/court-dashboard/pull/41/commits/4343c283e43e8bf5f14c5dfde07ee80446cf5357

File: src/components/Disputes/DisputeDetail.js [hunk 2]
```diff
@@ -10,19 +10,17 @@ import NoEvidence from './NoEvidence'
 
 const DisputeDetail = React.memo(function DisputeDetail({ match }) {
   const history = useHistory()
-  const disputes = useDisputesSubscription()
   const { id: disputeId } = match.params
 
-  const dispute = useMemo(
-    () => disputes.find(dispute => dispute.id === disputeId),
-    [disputeId, disputes]
+  const { dispute, fetching: disputeFetching } = useDisputeSubscription(
```
Comment (fabriziovigevani) [inline src/components/Disputes/DisputeDetail.js]: I think that for subscriptions, fetching is always true 😄

Comment (bpierre) [reply src/components/Disputes/DisputeDetail.js]: Not this one haha https://github.com/aragon/court-dashboard/pull/41/files#diff-4451f186da266821f8a6c366b69e3dc0R17

But I should probably use something else like `initialFetching` :thinking:

File: src/components/Disputes/hooks/useDisputeSubscription.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { useMemo } from 'react'
+import { useSubscription } from 'urql'
+import { reduceDispute } from '../reducer'
+import { SingleDispute } from '../../../queries/disputes'
+
+export default function useSingleDisputeSubscription(id) {
```
Comment (rperez89) [inline src/components/Disputes/hooks/useDisputeSubscription.js]: Should we use `useQuery` instead of useSubscription if we are directly querying by id?

Comment (bpierre) [reply src/components/Disputes/hooks/useDisputeSubscription.js]: I was thinking that a dispute might update its data, while we are on the page?

Comment (fabriziovigevani) [reply src/components/Disputes/hooks/useDisputeSubscription.js]: Yes, we will need a subscription here.

Comment (rperez89) [reply src/components/Disputes/hooks/useDisputeSubscription.js]: yeap totally true

**Pull request merged**